gradle/setup.gradle

```diff
@@ -112,10 +112,28 @@ publishing {
 			artifact javadocJar
 			//consider adding extra artifacts here, conditionally on submodule's name and perhaps in an afterEvaluate block
 			afterEvaluate {
-				if (project.name == 'reactor-core') {
-					artifact rootProject.tasks.docsZip
+				// Find the docs project (it's not a java project, so we need to lookup it)
+				def docsProject = project.findProject(':docs')
+				if (docsProject) {
+					// Access the docsZip task from the docs project
+					def docsZipTask = docsProject.tasks.findByName('docsZip')
+					if (docsZipTask) {
```

Comment:
Should there not be a failure if this task is not defined?

Reply:
first, there was a bug in this part: the docs zip should be included only in the `reactor-core` artifact, not in all artifacts. So, I added the missing check in the `afterEvaluate` section:

```
if (project.name == 'reactor-core') {
```
see https://github.com/reactor/reactor-core/pull/3806/commits/34c854e6ff3b2234a3bb57e858e33b24be719b36

Also, did this in this commit:
- if the current JDK version is greater than JDK8, then include the docsZip artifact as it was done previously, there is no need to change what was done before:
```
artifact docsProject.tasks.docsZip
```
- if the current JDK version is lower than JDK8, and if no docs zip archive is found from the docs/build/distributions/ directory, then add a WARN message to indicate to the user that the docs won't be published:
```
logger.warn("Antora docs zip not found from docs/ project. Use JDK 17+ to include docs in published artifacts.")
```
- finally, fixed and clarified comments